pex/environment.py

```diff
@@ -225,124 +263,244 @@ def __init__(
         # type: (...) -> None
         self._pex = pex
         self._pex_info = pex_info or PexInfo.from_pex(pex)
-        self._internal_cache = os.path.join(self._pex, self._pex_info.internal_cache)
-        self._activated = False
-        self._working_set = None
         self._interpreter = interpreter or PythonInterpreter.get()
-        self._inherit_path = self._pex_info.inherit_path
-        self._supported_tags = frozenset(self._interpreter.identity.supported_tags)
+
+        self._available_ranked_dists_by_key = defaultdict(
+            list
+        )  # type: DefaultDict[str, List[_RankedDistribution]]
+        self._activated_dists = None  # type: Optional[Iterable[Distribution]]
+
+        # The supported_tags come ordered most specific (platform specific) to least specific
+        # (universal). We want to rank most specific highest; so we need to reverse iteration order
+        # here.
+        self._supported_tags_to_rank = {
+            tag: rank
+            for rank, tag in enumerate(reversed(self._interpreter.identity.supported_tags))
```

Comment:
No, it guaranteed by `packaging.tags` so its not easy or wise to try to repro. The contract is PythonInterpreter / PythonIdentity grab their supported tags from packaging and do not disturb the ordering since it is vital.
https://github.com/pantsbuild/pex/blob/8d4f3db943ff0d3da47cd7225e73fb56bc0a99e5/pex/vendor/_vendored/packaging/packaging/tags.py#L833-L840